src/Swift.Bindings/src/TypeDatabase/ITypeDatabase.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Type database. Holds module databases and type records for Swift types.
+/// </summary>
+public interface ITypeDatabase
+{
+    /// <summary>
+    /// Loads a module database from a specified file.
+    /// </summary>
+    /// <param name="file">The file path of the module database to load.</param>
+    public void LoadModuleDatabaseFromFile(string file);
+
+    /// <summary>
+    /// Adds a module database to the type database.
+    /// </summary>
+    /// <param name="moduleDatabase">The module database to add.</param>
+    /// <exception cref="Exception">Thrown if a module with the same name already exists in the database.</exception>
+    public void AddModuleDatabase(IModuleDatabase moduleDatabase);
```

Comment:
I thought about it for some time. Maybe we should remove the `IModuleDatabase` abstraction and move the methods which add stuff to TypeDatabase out of this interface. This would make the `ITypeDatabase` fully readonly. Methods adding stuff to it could reside on concrete implementation and become implementation detail.

Curious to learn you thoughts.